### PR TITLE
Add credential Scope keyword 

### PR DESCRIPTION
**Pull request by rakshith91**

### CODE REVIEW

File: sdk/monitor/azure-monitor-query/azure/monitor/query/_metrics_query_client.py [hunk 1]
```diff
@@ -44,15 +44,22 @@ class MetricsQueryClient(object):
...
     """
 
     def __init__(self, credential, **kwargs):
         # type: (TokenCredential, Any) -> None
+        audience = kwargs.pop("audience", None)
         endpoint = kwargs.pop("endpoint", "https://management.azure.com")
+        if not endpoint.startswith("https://") and not endpoint.startswith("http://"):
+            endpoint = "https://" + endpoint
+        self._endpoint = endpoint
+
```
Comment (rakshith91) [inline sdk/monitor/azure-monitor-query/azure/monitor/query/_metrics_query_client.py:58]:
```suggestion

```

**Review by yunhaoling — APPROVED** [review]
LGTM, left some thoughts

File: sdk/monitor/azure-monitor-query/azure/monitor/query/aio/_metrics_query_client_async.py [hunk 2]
```diff
@@ -32,14 +32,20 @@ class MetricsQueryClient(object):
...
     :keyword endpoint: The endpoint to connect to. Defaults to 'https://management.azure.com'.
     :paramtype endpoint: str
+    :keyword audience: URL to use for credential authentication with AAD.
+    :paramtype audience: str
     """
 
     def __init__(self, credential: "AsyncTokenCredential", **kwargs: Any) -> None:
+        audience = kwargs.pop("audience", None)
         endpoint = kwargs.pop("endpoint", "https://management.azure.com")
+        if not endpoint.startswith("https://") and not endpoint.startswith("http://"):
```
Comment (KarishmaGhiya) [inline sdk/monitor/azure-monitor-query/azure/monitor/query/aio/_metrics_query_client_async.py:42]: why are both logic around "and" the same?

**Review by KarishmaGhiya — APPROVED** [review]

File: sdk/monitor/azure-monitor-query/azure/monitor/query/_logs_query_client.py [hunk 3]
```diff
@@ -49,15 +49,20 @@ class LogsQueryClient(object):
...
     """
 
     def __init__(self, credential, **kwargs):
         # type: (TokenCredential, Any) -> None
-
-        self._endpoint = kwargs.pop("endpoint", "https://api.loganalytics.io/v1")
+        audience = kwargs.pop("audience", None)
+        endpoint = kwargs.pop("endpoint", "https://api.loganalytics.io/v1")
+        if not endpoint.startswith("https://") and not endpoint.startswith("http://"):
+            endpoint = "https://" + endpoint
```
[the remark this answers is not in the document]
Comment (rakshith91) [reply sdk/monitor/azure-monitor-query/azure/monitor/query/_logs_query_client.py:61]: not really - it's more of a convenience

File: sdk/monitor/azure-monitor-query/README.md [hunk 4]
```diff
@@ -14,6 +14,10 @@ The Azure Monitor Query client library is used to execute read-only queries agai
 - [Samples][samples]
 - [Change log][changelog]
 
+## _Disclaimer_
```
[the remark this answers is not in the document]
Comment (rakshith91) [reply sdk/monitor/azure-monitor-query/README.md:17]: have that in a different PR :P

**Pull request merged**